### ISSUE TITLE
Maybe add a option to move the kernel somewhere else than just /boot?

### ISSUE THREAD

**Issue by zachzilla04** [Frogging-Family/linux-tkg#25]
Alright, so after further testing by actually building the kernel and installing it, despite changing the path where it was to be installed  in the Makefile, it still just installs in /boot.

**Comment by Berobad** [comment]
isn't the kernel nowadays in /lib/modules/ and mkinitcpio copies it to boot?

**Comment by zachzilla04** [comment]
I meant the vmlinuz-linux.

**Comment by zachzilla04** [comment]
And mkinitcpio doesn't copy it over to the place that i want it to.

**Comment by Tk-Glitch** [comment]
The current hooking in Arch would need to be bypassed. I'll see what I can do.

**Comment by Tk-Glitch** [comment]
oops

**Comment by damachine** [comment]
This is an older topic, but it might still be of interest.

I think the boot layout should be left to the user's system configuration rather than being handled by linux-tkg itself.

In my case, this was solved using `kernel-install` following the ArchWiki:
https://wiki.archlinux.org/title/Kernel-install

Hope this helps <3

**Comment by AdelKS** [comment]
is `kernel-install` used when making the official arch linux kernel packages ? and even then it will install in a fixed folder within the `.tar.zst` package 🤔

**Comment by damachine** [comment]
> is `kernel-install` used when making the official arch linux kernel packages ? 

No, not when building the official Arch kernel packages. kernel-install is a systemside mechanism used after the kernel is installed

> and even then it will install in a fixed folder within the `.tar.zst` package 🤔

Yeah, exactly. The package itself still has a fixed layout; kernel-install only comes into play on the installed system afterwards.

**Comment by AdelKS** [comment]
> Yeah, exactly. The package itself still has a fixed layout; kernel-install only comes into play on the installed system afterwards.

So it's a post-script in the package ? if that's the case, we should do that I suppose, not sure how _delete_ though

**Comment by damachine** [comment]
> So it's a post-script in the package ? if that's the case, we should do that I suppose, not sure how _delete_ though

Yeah, basically. There is already `[pacman-hook-kernel-install](https://aur.archlinux.org/packages/pacman-hook-kernel-install)` in the AUR doing exactly that. using pacman hooks to call kernel-install on install/remove. That might actually be a good reference for implementing it
